opensearch_py_ml/ml_commons/IAMRoleHelper.py

```diff
@@ -0,0 +1,301 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import json
+
+import boto3
+import requests
+from botocore.exceptions import ClientError
+
+
+class IAMRoleHelper:
+    """
+    Helper class for managing IAM roles and their interactions with OpenSearch.
+    """
+
+    def __init__(
+        self,
+        region,
+        opensearch_domain_url=None,
+        opensearch_domain_username=None,
+        opensearch_domain_password=None,
+        aws_user_name=None,
+        aws_role_name=None,
+        opensearch_domain_arn=None,
+    ):
+        """
+        Initialize the IAMRoleHelper with AWS and OpenSearch configurations.
+
+        :param region: AWS region.
+        :param opensearch_domain_url: URL of the OpenSearch domain.
+        :param opensearch_domain_username: Username for OpenSearch domain authentication.
+        :param opensearch_domain_password: Password for OpenSearch domain authentication.
+        :param aws_user_name: AWS IAM user name.
+        :param aws_role_name: AWS IAM role name.
+        :param opensearch_domain_arn: ARN of the OpenSearch domain.
+        """
+        self.region = region
+        self.opensearch_domain_url = opensearch_domain_url
+        self.opensearch_domain_username = opensearch_domain_username
+        self.opensearch_domain_password = opensearch_domain_password
+        self.aws_user_name = aws_user_name
+        self.aws_role_name = aws_role_name
+        self.opensearch_domain_arn = opensearch_domain_arn
+
+    def role_exists(self, role_name):
+        """
+        Check if an IAM role exists.
+
+        :param role_name: Name of the IAM role.
+        :return: True if the role exists, False otherwise.
+        """
+        iam_client = boto3.client("iam")
```

Comment:
Rather than having initializing the same client in every method, can we init this client in `__init__`? This is reduce the redundancy.